src/PdoConnection.php

```diff
@@ -238,7 +252,13 @@ public function begin()
      */
     public function rollBack()
     {
-        return $this->connect()->rollBack();
+        if (--$this->transactionCount === 0) {
```

Comment:
have you considered what happens if you call `rollBack()` before calling `begin()`? :smiley: